src/images/images.service.ts

```diff
@@ -68,6 +70,75 @@ export class ImagesService {
       );
     });
 
-    return { data };
+    return { images: data };
+  }
+
+  async searchImages(
+    searchQuery = '',
+    page = 1,
+    perPage: number,
+    userId?: number | string,
+  ): Promise<{
+    count: number;
+    page: number;
+    perPage: number;
+    images: Image[];
+  }> {
+    const params = searchQuery.split(' ');
+    const arr = permutationsOfArray(params.slice(0, 3));
+    const arrOfPromises: Promise<Image[]>[] = [];
+    const { limit, offset } = paginate(page, perPage);
+    const watermark = await this.watermarksRepository.findOneBy({
+      isDefault: true,
+    });
+
+    arr.forEach((query) => {
```

Comment:
This is not the best approach - we should use permutations to build WHERE/OR WHERE string and then use that in the query - and also with this cascading querying we cannot implement db level pagination...